group/assignment6/ParkingLot.java

```diff
@@ -0,0 +1,148 @@
+package assignment6;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+There is a parking lot with N spaces and N-1 cars in it. Your task is to write an algorithm 
+to rearrange the cars in a given way. Only one car can be moved at a time to the empty slot.
+*/
+public class ParkingLot {
+    /**
+    We have assummed cars have ids from 0 to N-2 and parking lots have ids from 0 to N-1. (both inclusive).
+    We represent the parking lot state as an array, where the content of each position p is the id of the car
+    situated on parking lot position p. An id of -1 represents an empty parking lot.
+    */
+    private int currentEmptySpotIndex = -1;
+    private int endEmptySpotIndex = -1;
+    private int[] parkingLotCurrent; //car in each position of the parking lot
+    private int[] parkingLotEnd;  // final desired state of the parking lot
+    private int[] carToPosition; //current position of each car
+    private int parkingLotSize;   
+    private List<Move> moves; 
+
+    public ParkingLot(int[] parkingLotCurrent, int[] parkingLotEnd) throws Exception {
+    	
+        checkValidInput(parkingLotCurrent, parkingLotEnd);
+         
+        this.parkingLotSize = parkingLotCurrent.length;
+        this.parkingLotCurrent = parkingLotCurrent;
+        this.parkingLotEnd = parkingLotEnd;
+        moves = new ArrayList<>();
```

Comment:
I think it would be better to leave just the previous 4 lines in the constructor itself and then do all the checks and the algorithm itself in a separate method. Usually, constructors should be simple, and just initiate the fields (I think).